skiko/src/jvmMain/kotlin/org/jetbrains/skiko/FrameDispatcher.kt

```diff
@@ -23,14 +22,15 @@ class FrameDispatcher(
         onFrame
     )
 
-    private var needFrame = CompletableDeferred<Unit>()
+    private var needFrame = false
+    private val channel = Channel<Unit>(Channel.CONFLATED)
 
     private val job = scope.launch {
         while (true) {
-            needFrame.await()
-            needFrame = CompletableDeferred()
+            if (!needFrame)
+                channel.receive()
```

Comment:
We have one more unnecessary frame. See `perform tasks scheduled in the frame after the frame`.

It is because we push value to the channel, but don't receive it on the second frame. We receive it only on the third frame.

Reply:
Yep, simplified that place.